### PR TITLE
Budget App

### PR DESCRIPTION
**Pull request by EzekielUtshudi**
**Hey Code Reviewer Kindly Enjoy reviewing this Milestone, On this PR I have implemented the following :**
### Splash screen

- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

### Sign up and login pages

- The user is  able to register in the app with full name, email and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

### Home page (catagories page)

- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, IMG & the total amount of all the transactions that belong to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

### Transactions page

- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page, the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

"### Add a new category" page

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

### "Add a new transaction" page

- The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.
- Technical requirements
- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.

### CODE REVIEW

File: spec/features/users_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  scenario 'show groups_path' do
+    expect(page).to have_content('Log in')
+  end
+
+  scenario 'show groups_path' do
+    expect(page).to have_content('Sign up')
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline spec/features/users_index_spec.rb:15]: - [ ] Some of your tests are failing, Kindly fix this
 
Screenshot from 2022-08-05 21-58-34

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @EzekielUtshudi , 
## STATUS: REQUIRED CHANGES :recycle: 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/features/users_index_spec.rb (hunk 1 above)
Comment (EzekielUtshudi) [reply spec/features/users_index_spec.rb:15]: Thanks man for the feedback,I fixed that, and below is the screenshot

image

**Review by Deepakdanger — APPROVED** [review]
Hi @EzekielUtshudi, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**